Update Pagination Logic on Hubs

manveerxyz opened the pull request:
Had to make some changes to the backend to make pagination work properly.

- Ensure the `ordering` param is actually getting used
- Add "id" as an ordering field to break ties for hubs that have identical scores
- Added tests for ordering and pagination

lightninglu10 commented on src/hub/filters.py:
```diff
@@ -40,7 +40,7 @@ def filter(self, qs, value):
             )
 
-            qs = qs.annotate(score=score).order_by("-score")
+            qs = qs.annotate(score=score).order_by(*value, "id")
```
so we're ordering first by score and then by id?

manveerxyz replied:
`id` is meant to be a tie-breaker if the scores are the same. This change is to resolve the issue we had with the same item showing up on multiple pages.

manveerxyz commented:
Okay, closed https://github.com/ResearchHub/researchhub-backend/pull/1381 and removed sorting filters in this PR.

Also updated the caching in the view.

lightninglu10 commented:
Looks good, going to merge & deploy
